components/fxa-client/src/device.rs

```diff
@@ -9,17 +9,24 @@ use crate::{
     commands,
     error::*,
     http_client::{
-        CommandData, DeviceUpdateRequest, DeviceUpdateRequestBuilder, PendingCommand,
-        UpdateDeviceResponse,
+        DeviceUpdateRequest, DeviceUpdateRequestBuilder, PendingCommand, UpdateDeviceResponse,
     },
-    util, CachedResponse, FirefoxAccount, IncomingDeviceCommand,
+    telemetry, util, CachedResponse, FirefoxAccount, IncomingDeviceCommand,
 };
 use serde_derive::*;
 use std::collections::{HashMap, HashSet};
 
 // An devices response is considered fresh for `DEVICES_FRESHNESS_THRESHOLD` ms.
 const DEVICES_FRESHNESS_THRESHOLD: u64 = 60_000; // 1 minute
 
+/// The reason we are fetching commands.
+pub enum CommandFetchReason {
```

Comment:
This is a pretty small enum—might as well derive `Clone` and `Copy` on it, and pass it by value everywhere.

Reply:
👍 - but I remember going backwards and forwards about this and landed on "if it's trivial to pass by ref, then why not?" - but yeah, it probably does read a little better, so I did it.